### DISCUSSION TITLE
How can I merge several graphs into a giant graph?

### DISCUSSION THREAD

**Discussion by chenyuqi990215** [pyg-team/pytorch_geometric#3442] [Q&A] [answered]
I have a graph with edge_index shape `[2, e]`, and I would like to construct a batched graph with the same edge_index to get a giant graph with edge_index shape `[2, b*e]`, where `b` is the batch_size.

The function may look like the following:

```
def merge_graphs(edge_index, b)
```

Is there any simple solution to the function?

**Comment by rusty1s** [comment, marked as answer]
It's probably the easiest way to wrap them int `Data` objects:
```python
data_list = [Data(edge_index=edge_index), ...]
edge_index = Batch.from_data_list(data_list).edge_index
```
